Assets/Project/Runtime/_Scripts/HealthSystem/HealthBar.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Project
+{
+    [RequireComponent(typeof(Slider))]
+    public class HealthBar : MonoBehaviour
+    {
+        private Slider healthBar;
+        [SerializeField] private GameObject objectWithHealth;
+        private IHasHealth hasHealth;
+
+        void Start()
+        {
+            healthBar = GetComponent<Slider>();
+
+            if (objectWithHealth.TryGetComponent<IHasHealth>(out hasHealth)){
+                hasHealth.OnHealthChanged += UpdateHealthBar;
+            }
+            else {Debug.Log("No IHasHealth found on GameObject");}
```

Comment:
```suggestion
            if (objectWithHealth.TryGetComponent<IHasHealth>(out ref hasHealth)){
                hasHealth.OnHealthChanged += UpdateHealthBar;
            }
            else {Debug.Log("No IHasHealth found on GameObject");}
```

Reply:
Beter als het in de `OnEnable()` methode staat.

Reply:
Waarom die var? Dan kan ik die hasHealth enkel gebruiken in de scope van OnEnable(), maar ik heb die nog nodig in de UpdateHealthBar().

Reply:
Ah juist dat had ik niet gezien, ik was te vlug.